engine/src/main/java/org/hibernate/validator/internal/util/annotationfactory/AnnotationProxy.java

```diff
@@ -176,6 +182,18 @@ else if ( m.getDefaultValue() != null ) {
 		return result;
 	}
 
+	private Map<String, Object> getAnnotationValues(Annotation annotation) {
+		if ( Proxy.isProxyClass( annotation.getClass() ) ) {
+			InvocationHandler invocationHandler = Proxy.getInvocationHandler( annotation );
+			if ( invocationHandler instanceof AnnotationProxy ) {
```

Comment:
The commit message is a bit mis-leading IMHO, as it's only about proxies created by ourselves. I first thought you'd access the parameter map of the actual `sun.reflect.annotation.AnnotationInvocationHandler`.